api/v1alpha1/siteconfig_types.go

```diff
@@ -0,0 +1,448 @@
+
+	// +optional
+	SSHPrivateKeySecretRef SSHPrivateKeySecretRef `json:"sshPrivateKeySecretRef,omitempty"`
+
+	// BaseDomain is the base domain to use for all clusters created in this pool.
+	// +required
+	BaseDomain string `json:"baseDomain,omitempty"`
+
+	// APIVIP is the virtual IP used to reach the OpenShift cluster's API.
+	// +optional
+	ApiVIP string `json:"apiVIP,omitempty"`
+
+	// APIVIPs are the virtual IPs used to reach the OpenShift cluster's API.
+	// Enter one IP address for single-stack clusters, or up to two for dual-stack clusters (at
+	// most one IP address per IP stack used). The order of stacks should be the same as order
+	// of subnets in Cluster Networks, Service Networks, and Machine Networks.
+	// +kubebuilder:validation:MaxItems=2
+	// +optional
+	ApiVIPs []string `json:"apiVIPs,omitempty"`
+
+	// IngressVIP is the virtual IP used for cluster ingress traffic.
+	// +optional
+	IngressVIP string `json:"ingressVIP,omitempty"`
+
+	// IngressVIPs are the virtual IPs used for cluster ingress traffic.
+	// Enter one IP address for single-stack clusters, or up to two for dual-stack clusters (at
+	// most one IP address per IP stack used). The order of stacks should be the same as order
+	// of subnets in Cluster Networks, Service Networks, and Machine Networks.
+	// +kubebuilder:validation:MaxItems=2
+	// +optional
+	IngressVIPs []string `json:"ingressVIPs,omitempty"`
+
+	// HoldInstallation will prevent installation from happening when true.
+	// Inspection and validation will proceed as usual, but once the RequirementsMet condition is true,
+	// installation will not begin until this field is set to false.
+	// +kubebuilder:default:=false
+	// +optional
+	HoldInstallation bool `json:"holdInstallation,omitempty"`
+
+	// AdditionalNTPSources is a list of NTP sources (hostname or IP) to be added to all cluster
+	// hosts. They are added to any NTP sources that were configured through other means.
+	// +optional
+	AdditionalNTPSources []string `json:"additionalNTPSources,omitempty"`
+
+	// MachineNetwork is the list of IP address pools for machines.
+	// +optional
+	MachineNetwork []MachineNetworkEntry `json:"machineNetwork,omitempty"`
+
+	// ClusterNetwork is the list of IP address pools for pods.
+	// +optional
+	ClusterNetwork []ClusterNetworkEntry `json:"clusterNetwork,omitempty"`
+
+	// ServiceNetwork is the list of IP address pools for services.
+	// +optional
+	ServiceNetwork []string `json:"serviceNetwork,omitempty"`
+
+	// NetworkType is the Container Network Interface (CNI) plug-in to install
+	// The default value is OpenShiftSDN for IPv4, and OVNKubernetes for IPv6 or SNO
+	// +kubebuilder:validation:Enum=OpenShiftSDN;OVNKubernetes
+	// +kubebuilder:default:=OVNKubernetes
+	// +optional
+	NetworkType string `json:"networkType,omitempty"`
+
+	// Additional cluster-wide annotations to be applied to the rendered templates
+	// +optional
+	CrAnnotations CrAnnotations `json:"crAnnotations,omitempty"`
```

Comment:
I don't see how we are going to handle node deletion without having `crSuppression` field. Currently, to delete a CR, for example the BMH CR which gets deleted only after it gets the annotation marked as "to be deleted". [Link](https://github.com/openshift-kni/cnf-features-deploy/blob/master/ztp/gitops-subscriptions/argocd/WorkerNodeDeletion.md)

Reply:
I don’t think we would need the `crSuppression` field to perform node deletion with the new SiteConfig design (i.e. decoupled templates from the SiteConfig API). To delete a node, we would apply the annotation `bmac.agent-install.openshift.io/remove-agent-and-node-on-delete: true` via `SiteConfig` in the first pass, and then once the BMH resource has the annotation applied, on the second pass, a new template configmap consisting of a set of node-related templates (which would exclude the BMH template) can be referenced and applied. This should have the same effect as `crSuppression`. 

We had to introduce the `crSuppression` concept in the original siteconfig plugin as there was no other way to specify which templates to render, and which to exclude. This problem is now taken care of by the introduction of the `templateRefs` fields at both cluster and node levels. 

Having said this, we can still support the concept of `crSuppression` and "ignore" templates in the referenced configmap which are listed in `crSuppression`. This would have a noop effect if the referenced template configmap(s) do not have the CR listed in `crSuppression`.

Reply:
I have introduced the field `SuppressedManifests` at both cluster-level and node-level.